File: Lab_9/Main.py
# Excepcion para depositos invalidos
class DepositoInvalidoException(Exception):
    def __init__(self, mensaje="El monto a depositar no puede ser negativo."):
        super().__init__(mensaje)

# Excepcion para retiros invalidos
class RetiroInvalidoException(Exception):
    def __init__(self, mensaje="El monto a retiro no puede ser negativo."):
        super().__init__(mensaje)

# Excepcion para fondos insuficientes
class FondosInsuficientesException(Exception):
    def __init__(self, mensaje="Fondos insuficientes para realizar el retiro."):
        super().__init__(mensaje)

# Clase cuenta bancaria
class cuentaBancaria:
    def __init__(self, numeroCuenta, saldo):
        self.numeroCuenta = numeroCuenta
        self.saldo = saldo

    def depositar(self, cantidad):
        if cantidad<0:
            raise DepositoInvalidoException()
        self.saldo += cantidad

    def retirar(self, cantidad):
        if cantidad<0:
            raise RetiroInvalidoException()
        if cantidad>self.saldo:
            raise FondosInsuficientesException()
        self.saldo -= cantidad

    def getSaldo(self):
        return self.saldo
    
    def getCuenta(self):
        return self.numeroCuenta
    
# Programa principal
def main():
    num = input("Ingresa el numero de cuenta: ")
    saldo = float(input("Ingresa el saldo inicial: "))
    cuenta = cuentaBancaria(num, saldo)

    while True:
        print("\n--- Menu ---")
        print("1. Depositar dinero")
        print("2. Retirar dinero")
        print("3. Consultar saldo")
        print("4. Salir")

        try:
            opc = int(input("Seleccione una opcion: "))
            if opc == 1:
                cantidad = float(input("Ingrese la cantidad a depositar: "))
                cuenta.depositar(cantidad)
                print(f"Deposito exitoso. Saldo actual: ${cuenta.getSaldo():.2f}")
            elif opc == 2:
                cantidad = float(input("Ingrese la cantidad a retirar: "))
                cuenta.retirar(cantidad)
                print(f"Retirar exitoso. Saldo actual: ${cuenta.getSaldo():.2f}")
            elif opc == 3:
                print(f"Saldo actual: ${cuenta.getSaldo():.2f}")
            elif opc == 4:
                print("Hasta luego.")
                break
            else:
                print("Opcion no valida. Intente de nuevo.")
        except ValueError:
            print("Error: Por favor, ingrese un numero valido.")
        except DepositoInvalidoException as e:
            print(f"Error: {e}")
        except RetiroInvalidoException as e:
            print(f"Error: {e}")
        except FondosInsuficientesException as e: 
            print(f"Error: {e}")

if __name__ == "__main__":
    main()

File: Lab_9/Main.java
import java.util.Scanner;

// Excepcion para depositos invalidos
class DepositoInvalidoException extends Exception{
    public DepositoInvalidoException(String mensaje){
        super(mensaje);
    }
}

// Excepcion para retiros invalidos
class RetiroInvalidoException extends Exception{
    public RetiroInvalidoException(String mensaje){
        super(mensaje);
    }
}

// Excepcion para fondos insuficientes
class FondosInsuficientesException extends Exception{
    public FondosInsuficientesException(String mensaje){
        super(mensaje);
    }
}

// Clase Cuenta Bancaria
class cuentaBancaria{
    private String numeroCuenta;
    private double saldo;
    
    public cuentaBancaria(String numeroCuenta, double saldo){
        this.numeroCuenta = numeroCuenta;
        this.saldo = saldo;
    }
    
    public void depositar(double cantidad) throws DepositoInvalidoException{
        if (cantidad<0){
            throw new DepositoInvalidoException("El monto a depositar no puede ser negativo.");
        }
        saldo += cantidad;
    }
    
    public void retirar(double cantidad) throws RetiroInvalidoException, FondosInsuficientesException{
        if (cantidad<0){
            throw new RetiroInvalidoException("El monto a retirar no puede ser negativo.");
        }
        if (cantidad>saldo){
            throw new FondosInsuficientesException("Fondos insuficientes para realizar el retiro.");
        }
        saldo -= cantidad;
    }
    
    public double getSaldo(){
        return saldo;
    }
    
    public String getnumeroCuenta(){
        return numeroCuenta;
    }
}

// Programa Principal
public class Main {
    public static void main(String[] args) {
        Scanner lector = new Scanner(System.in);
        System.out.println("Ingrese el numero de cuenta: ");
        String num = lector.nextLine();
        System.out.println("Ingrese el saldo inicial: ");
        double saldo = lector.nextDouble();
        
        cuentaBancaria cuenta = new cuentaBancaria(num, saldo);
        boolean s = true;
        
        while(s){
            System.out.println("\n--- Menu ---");
            System.out.println("1. Depositar dinero");
            System.out.println("2. Retirar dinero");
            System.out.println("3. Consultar saldo");
            System.out.println("4. Salir");
            System.out.println("Seleccione una opcion: ");
            int opc = lector.nextInt();
            
            try{
                switch(opc){
                    case 1:
                        System.out.println("Ingrese la cantidad a depositar: ");
                        double deposito = lector.nextDouble();
                        cuenta.depositar(deposito);
                        System.out.println("Deposito exitoso. Saldo actual: $"+cuenta.getSaldo());
                        break;
                    case 2:
                        System.out.println("Ingrese la cantidad a retirar: ");
                        double retiro = lector.nextDouble();
                        cuenta.retirar(retiro);
                        System.out.println("Retiro exitoso. Saldo actual: $"+cuenta.getSaldo());
                        break;
                    case 3:
                        System.out.println("Saldo actual: $"+cuenta.getSaldo());
                        break;
                    case 4:
                        s = false;
                        System.out.println("Hasta luego.");
                        break;
                    default:
                        System.out.println("Opcion no valida. Intente de nuevo.");
                }
            } catch (DepositoInvalidoException e){
                System.out.println("Error: "+e.getMessage());
            } catch (RetiroInvalidoException e){
                System.out.println("Error: "+e.getMessage());
            } catch (FondosInsuficientesException e){
                System.out.println("Error: "+e.getMessage());
            }
        }
        
        lector.close();
    }
}